backend/crud.py

```diff
@@ -197,14 +261,235 @@ def update_schema(db: Session, schema_id: int, data: SchemaUpdateSchema) -> Sche
+
+    attr_groups: Dict[str, List[Attribute]] = defaultdict(list)
+    attributes = [i.attribute for i in attrs_to_include]
+    for attr in attributes:
+        attr_groups[attr.type.name].append(attr)
+    
+    ent_ids = [i.id for i in entities]
+    attr_map = {i.attribute.id: i.attribute.name for i in attrs_to_include}
+    for group, attrs in attr_groups.items():
+        value_model: Value = AttrType[group].value.model
+        q = (
+            select(value_model)
+            .where(value_model.entity_id.in_(ent_ids))
+            .where(value_model.attribute_id.in_([i.id for i in attrs]))
+        )
+        rows = db.execute(q).scalars().all()
+        for r in rows:
+            ent: dict = results_map[r.entity_id]
+            attr: str = attr_map[r.attribute_id]
+            if isinstance(ent[attr], list):
+                ent[attr].append(r.value)
+            else:
+                ent[attr] = r.value
+    
+    results = list(results_map.values())  
+    return results
+
+
+FILTER_MAP = {
+    'eq': '__eq__',
+    'lt': '__lt__',
+    'gt': '__gt__',
+    'le': '__le__',
+    'ge': '__ge__',
+    'ne': '__ne__',
+    'contains': 'contains',
+    'regexp': 'regexp_match'
+}
+
+ALLOWED_FILTERS = {
+    AttrType.STR: ['eq', 'lt', 'gt', 'le', 'ge', 'ne', 'contains', 'regexp'],
+    AttrType.INT: ['eq', 'lt', 'gt', 'le', 'ge', 'ne'],
+    AttrType.FLOAT: ['eq', 'lt', 'gt', 'le', 'ge', 'ne'],
+    AttrType.DT: ['eq', 'lt', 'gt', 'le', 'ge', 'ne'],
+    AttrType.BOOL: ['eq', 'ne']
+}
+
+
+def _parse_filters(filters: dict, attrs: List[str]) -> Tuple[Dict[str, Dict[str, Any]], Dict[str, Dict[str, Any]]]:
+    '''Returns tuple of two `dict`s like `{attr_name: {op1: value, op2: value}}`.
+    First `dict` is for attribute filters, second is for `Entity.name` filters
+    '''
+    attrs_filters = defaultdict(dict)
+    name_filters = {}
+    for f, v in filters.items():
+        split = f.rsplit('.', maxsplit=1)
+        attr = split[0]
+        filter = 'eq' if len(split) == 1 else split[-1]
+        if attr == 'name':
+            name_filters[filter] = v
+            continue
+        elif attr not in attrs:
+            raise InvalidFilterAttributeException(attr=attr, allowed_attrs=attrs)
+        elif filter not in FILTER_MAP:
+            raise InvalidFilterOperatorException(attr=attr, filter=filter)
+        attrs_filters[attr][FILTER_MAP[filter]] = v
+    
+    name_filters = {k: v for k,v in name_filters.items() if k in FILTER_MAP and k in ALLOWED_FILTERS[AttrType.STR]}
+    return attrs_filters, name_filters
+
+
+def _query_entity_with_filters(filters: dict, schema: Schema, all: bool = False, deleted_only: bool = False) -> CompoundSelect:
+    '''Returns intersection query of several queries with filters
+    to get entities that satisfy all conditions from `filters`
+    '''
+    
+    attrs = {i.attribute.name: i.attribute for i in schema.attr_defs if not i.list and i.attribute.type in ALLOWED_FILTERS}
```

Comment:
If I'm reading this correctly, we are excluding all list-type attributes from filtering. I don't think we should do that :thinking:

Reply:
It seems I was already overlooking this limitation in `dynamic_routes._filters_request_model`.

Reply:
Yes, I prohibited filtering on listed fields because... I don't actually remember why🙄 If you think that we should be able to filter on lists as well (when we check all values for provided condition) then I'll take a look later to see if it's going to work well if I just remove that `if`

Reply:
> If you think that we should be able to filter on lists 

Yes, I think this would make sense :grin: 

> when we check all values for provided condition

This might be counter-intuitive for users. I believe it would be more intuitive, if we just used the existing query for lists as well. In other words, on lists it's sufficient that any value in the list fulfills the filter condition(s).

> see if it's going to work well if I just remove that `if`

Thank you. If it does not go that well, we can address this in a separate PR.

Reply:
> it's sufficient that any value in the list fulfills the filter condition

Yeah, that was my point as well, I just meant that we'll have to check many values from list until we (don't) find satisfying our condition. And that also leads to question: what filters do we use in listed attrs? Just the ones which apply to underlying data type? i.e. if we have list of strings then for this attribute we can use all filters which we can use for string attributes

Reply:
So, I've allowed filtering lists and that's how it works now. Consider we have Jack and Jane with attr `fav_color` equal to `['red', 'blue']` and `['red', 'black']` correspondingly. We'll get such results for following queries:
* `fav_color.contains=b` --> Jack, Jane (`blue` and `black`)
* `fav_color.eq=black` --> Jane
* `fav_color.ne=black` --> Jack, Jane (Jane has `black`, but also has `red` which satisfies our condition)

I don't like `not equal` works in this case, but I guess it the most we can get without writing logic specifically for lists

Reply:
> I don't like `not equal` works in this case

Yes, I can see why :joy: I'd say we keep it like that and we address this via a separate issue and PR.

Reply:
Created #27.